posts/convertMarkdownToPost.ts

```diff
@@ -0,0 +1,47 @@
+import remark from 'remark';
+import html from 'remark-html';
+import frontmatter from 'remark-frontmatter';
+import extract from 'remark-extract-frontmatter';
+import YAML from 'yaml';
+
+async function convertMarkdownToPost(markdown: string): Promise<Post> {
+  const vFile = await parseMarkdownWithFrontmatter(markdown);
+
+  const { title } = extractPostMetadata(vFile.data);
+
+  return {
+    content: vFile.toString(),
+    metadata: {
+      title,
+    },
+  };
+}
+
+/**
+ * @private
+ *
+ * parses md file and returns vFile
+ */
+function parseMarkdownWithFrontmatter(markdown: string) {
+  return remark()
+    .use(frontmatter, ['yaml'])
+    .use(extract, { yaml: YAML.parse })
+    .use(html)
+    .process(markdown);
+}
+
+/**
+ * @private
+ */
+function extractPostMetadata(data: any) {
+  let title = '';
+  if ('title' in data) {
+    title = data.title;
+  }
+
+  return {
+    title,
+  };
+}
```

Comment:
Out of curiosity – don't you prefer a more declarative approach to this part? Especially since it's such a simple piece of logic. I.e.:
```ts
function extractPostMetadata(data: any) {
  return {
    title: 'title' in data ? data.title : '',
    // or leveraging the null coalescing operator if you want to filter out both `null` and `undefined`
    title: data.title ?? ''
  };
}
```

Reply:
yes, declarative ftw! thanks